misk/src/main/kotlin/misk/web/actions/reflect.kt

```diff
@@ -61,3 +62,45 @@ internal fun Class<*>.superclasses(): Set<Class<*>> {
   }
   return result
 }
+
+/**
+ * Attempts to find a non-synthetic method with the same declaring class, name, and parameters. If
+ * this is already non-synthetic it is returned immediately. If no non-synthetic method is found
+ * this method is returned.
+ *
+ * ### Synthetic Methods?
+ *
+ * Sometimes the compiler generates multiple methods with the same name and parameters, but
+ * different return types. This is caused by covariant return types. Consider this class:
+ *
+ * ```
+ * class PointIterator implements Iterator<Point> {
+ *   override boolean hasNext() { ... }
+ *   override Point next() { ... }
+ * }
+ * ```
+ *
+ * When compiled, javac creates a synthetic `next()` method that returns `Object` to conform to the
+ * `Iterator` interface:
+ *
+ * ```
+ * public boolean hasNext()
+ * public Point next()
+ * public synthetic bridge Object next()
+ * ```
+ *
+ * We prefer non-synthetic methods for reflection because that's where annotations will be.
+ */
+internal fun Method.preferNonSynthetic(): Method {
```

Comment:
nit: This name reads to me like it should have a boolean return type, would a better name be `nonSyntheticIfAvailable`? I don't think that's much better though TBH :)